ipinfo/cmd_asn.go

```diff
@@ -1,113 +1,87 @@
 package main
 
 import (
-	"errors"
 	"fmt"
-	"net/http"
+	"github.com/ipinfo/cli/lib"
+	"github.com/spf13/pflag"
+	"os"
 
-	"github.com/fatih/color"
 	"github.com/ipinfo/cli/lib/complete"
 	"github.com/ipinfo/cli/lib/complete/predict"
-	"github.com/ipinfo/go/v2/ipinfo"
-	"github.com/spf13/pflag"
 )
 
 var completionsASN = &complete.Command{
+	Sub: map[string]*complete.Command{
+		"bulk": completionsASNBulk,
+	},
 	Flags: map[string]complete.Predictor{
-		"-t":        predict.Nothing,
-		"--token":   predict.Nothing,
-		"--nocache": predict.Nothing,
-		"-h":        predict.Nothing,
-		"--help":    predict.Nothing,
-		"-f":        predict.Set(asnFields),
-		"--field":   predict.Set(asnFields),
-		"--nocolor": predict.Nothing,
-		"-p":        predict.Nothing,
-		"--pretty":  predict.Nothing,
-		"-j":        predict.Nothing,
-		"--json":    predict.Nothing,
-		"-c":        predict.Nothing,
-		"--csv":     predict.Nothing,
+		"-h":     predict.Nothing,
+		"--help": predict.Nothing,
 	},
 }
 
-func printHelpASN(asn string) {
+func printHelpASN() {
+
 	fmt.Printf(
-		`Usage: %s %s [<opts>]
+		`Usage: %s asn <cmd> [<opts>] 
+
+Commands:
+  bulk        lookup ASNs in bulk
 
 Options:
   General:
-    --token <tok>, -t <tok>
-      use <tok> as API token.
-    --nocache
-      do not use the cache.
     --help, -h
       show help.
-
-  Outputs:
-    --field <field>, -f <field>
-      lookup only specific fields in the output.
-      field names correspond to JSON keys, e.g. 'registry' or 'allocated'.
-      multiple field names must be separated by commas.
-    --nocolor
-      disable colored output.
-
-  Formats:
-    --json, -j
-      output JSON format. (default)
-    --yaml, -y
-      output YAML format.
-`, progBase, asn)
+`, progBase)
 }
 
-func cmdASN(asn string) error {
-	var fTok string
-	var fField []string
-	var fJSON bool
-	var fYAML bool
-
-	pflag.StringVarP(&fTok, "token", "t", "", "the token to use.")
-	pflag.BoolVar(&fNoCache, "nocache", false, "disable the cache.")
-	pflag.BoolVarP(&fHelp, "help", "h", false, "show help.")
-	pflag.StringSliceVarP(&fField, "field", "f", nil, "specific field to lookup.")
-	pflag.BoolVarP(&fJSON, "json", "j", true, "output JSON format. (default)")
-	pflag.BoolVarP(&fYAML, "yaml", "y", false, "output YAML format.")
-	pflag.BoolVar(&fNoColor, "nocolor", false, "disable color output.")
-	pflag.Parse()
-
-	if fNoColor {
-		color.NoColor = true
+// cmdASN is the handler for the "asn" command.
+func cmdASN() error {
+	var err error
+	cmd := ""
+	if len(os.Args) > 2 {
+		cmd = os.Args[2]
 	}
 
-	if fHelp {
-		printHelpASN(asn)
-		return nil
-	}
+	switch {
+	case cmd == "bulk":
+		err = cmdASNBulk()
+	default:
```

Comment:
This logic should ideally be in a `cmdASNDefault` handler, same as we have at the root.